src/yb/integration-tests/load_balancer_placement_policy-test.cc

```diff
@@ -0,0 +1,336 @@
+    external_mini_cluster()->master(),
+    external_mini_cluster()->tablet_server(4)));
+  ASSERT_OK(external_mini_cluster()->AddTServerToBlacklist(
+    external_mini_cluster()->master(),
+    external_mini_cluster()->tablet_server(5)));
+  WaitForLoadBalancer();
+
+  num_tservers -= 2;
+  GetLoadOnTservers(custom_policy_table, num_tservers, &counts_per_ts);
+  // ts0 in c.r.z0 should have no tablets in it.
+  ASSERT_EQ(counts_per_ts[0], 0);
+  // The other tablet servers should have tablets spread equally.
+  ASSERT_EQ(counts_per_ts[1], counts_per_ts[2]);
+  ASSERT_EQ(counts_per_ts[2], counts_per_ts[3]);
+
+  // The table with cluster placement policy should continue to have tablets spread across all
+  // tservers.
+  GetLoadOnTservers(table_name().table_name(), num_tservers, &counts_per_ts);
+  for (int ii = 0; ii < num_tservers; ++ii) {
+    ASSERT_GT(counts_per_ts[ii], 0);
+  }
+}
+
+TEST_F(LoadBalancerPlacementPolicyTest, AlterPlacementDataConsistencyTest) {
+  // Set cluster placement policy.
+  ASSERT_OK(yb_admin_client_->ModifyPlacementInfo("c.r.z0,c.r.z1", 2, ""));
+
+  // Start workload on a table.
+  const string& table = "placement-data-consistency-test";
+  const yb::client::YBTableName placement_table(
+    YQL_DATABASE_CQL, table_name().namespace_name(), table);
+
+  TestWorkload workload(external_mini_cluster());
+  workload.set_table_name(placement_table);
+  workload.Setup();
+  workload.Start();
+
+  // Change its placement policy such that it now has additional replicas spanning additional
+  // tservers.
+  ASSERT_OK(yb_admin_client_->ModifyTablePlacementInfo(placement_table, "c.r.z0,c.r.z1,c.r.z2", 3, ""));
+  WaitForLoadBalancer();
+
+  // Verify that the placement policy is honored.
+  vector<int> counts_per_ts;
+  GetLoadOnTservers(table, num_tablet_servers(), &counts_per_ts);
+  for (int ii = 0; ii < 3; ++ii) {
+    ASSERT_EQ(counts_per_ts[ii], 1);
+  }
+
+  // Change placement policy such that it now spans lesser replicas spanning fewer tservers.
+  ASSERT_OK(yb_admin_client_->ModifyTablePlacementInfo(placement_table, "c.r.z0", 1, ""));
+  WaitForLoadBalancer();
+
+  // Verify that placement policy is honored.
+  GetLoadOnTservers(table, num_tablet_servers(), &counts_per_ts);
+  // The table is RF1 and confined to zone 0. Ts0 should have 1 tablet.
+  // The other two tablet servers should not have any tablets.
+  ASSERT_EQ(counts_per_ts[0], 1);
+  ASSERT_EQ(counts_per_ts[1], 0);
+  ASSERT_EQ(counts_per_ts[2], 0);
+
+  // Verify that the data inserted is still sane.
+  workload.StopAndJoin();
+  int rows_inserted = workload.rows_inserted();
+  LOG(INFO) << "Number of rows inserted: " << rows_inserted;
+
+  // Verify that number of rows is as expected.
+  ClusterVerifier cluster_verifier(external_mini_cluster());
+  ASSERT_NO_FATALS(cluster_verifier.CheckCluster());
+  ASSERT_NO_FATALS(cluster_verifier.CheckRowCount(
+    placement_table, ClusterVerifier::EXACTLY, rows_inserted));
+}
+
```

Comment:
Not sure if its already tested, but I would add/modify a test that calls ModifyPlacementInfo with the optional placement_uuid set (3rd argument) and make sure table level overrides still work.

Reply:
Done